### PR TITLE
use travis for mima and whitesource, remain no jenkins for tests (in parallel)

### PR DESCRIPTION
**Pull request by ktoso**
Mirror travis config from akka-http, we have pretty much the same needs here

### CODE REVIEW

**Review by raboof — COMMENTED** [review]
Would this be intended to replace the jenkins PR validation? We do a bit more there, right? Might be nice to have that in Travis (perhaps we can more easily parallelize some things there as well), let's see if it can handle a project of this size though.

**Review by raboof — APPROVED** [review]

**Comment by ktoso** [comment]
I encrypted the whitesource key, should not also perform whitesource checks then

**Comment by ktoso** [comment]
Should now pass on travis I believe...

**Comment by ktoso** [comment]
Some second pair of eyes plz?

**Review by johanandren — APPROVED** [review]
Cool. LGTM (given that it works ofc)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+language: scala
+
+sudo: false
+
+scala:
+  - "2.12.4"
```
Comment (patriknw) [inline .travis.yml]: 2.12.5?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt/boot
+
+script:
+  # need to override as the default is to test
+  - sbt -J-XX:ReservedCodeCacheSize=256m ++$TRAVIS_SCALA_VERSION update mimaReportBinaryIssues
+  # make 'git branch' work again
+  - git branch -f "$TRAVIS_BRANCH" && git checkout "$TRAVIS_BRANCH"
+  # check policies, if on master also upload
+  - if [[ "$TRAVIS_SECURE_ENV_VARS" == "true" ]]; then if [[ "$TRAVIS_BRANCH" == "master" && "$TRAVIS_EVENT_TYPE" == "push" ]]; then sbt 'set credentials += Credentials("whitesource", "whitesourcesoftware.com", "", System.getenv("WHITESOURCE_KEY"))' whitesourceCheckPolicies whitesourceUpdate; else sbt 'set credentials += Credentials("whitesource", "whitesourcesoftware.com", "", System.getenv("WHITESOURCE_KEY"))' whitesourceCheckPolicies; fi ; fi
```
Comment (patriknw) [inline .travis.yml]: do we also run this as part of one of the nightly job, as an extra check "before" releasing?

File: .travis.yml (hunk 1 above)
Comment (ktoso) [reply .travis.yml]: Thanks

File: .travis.yml (hunk 2 above)
Comment (ktoso) [reply .travis.yml]: Yes it still is in nightlies

Comment (ktoso) [reply .travis.yml]: so whitesource is in: travis on pr validation, runs in parallel to jenkins tests, can be restarted individually, and then in nightlies

**Pull request merged**